File: practice_ruby/ruby_codetest_2/every_nth.rb
# parameters are sentence, n & offset
# break sentence into 2 parts at the offset
# get every nth word from the 2nd parts
# put the 2 parts together

def every_nth(sentence, n, offset)
  words = sentence.split(" ")
  first_part = words[0..offset - 1]
  last_part = words[offset..-1]
  every_nth = []

  last_part.each_with_index do |word, i|
    every_nth << word if ((i + 1) % n == 0) && (i > 0)
  end
  
  puts (first_part + every_nth).join(" ")
  (first_part + every_nth).join(" ")
end

every_nth("This is a really long sentence with a lot of words in it", 2, 3)
every_nth("Here is another one to try out", 3, 1)

File: javascript_practice/test/codetest/everyNth/everyNth.js
function capitalizeEveryNthWord(sentence, offset, n) {
        var words = sentence.split(" ");
        var firstPart = [];
        var lastPart = words;

        if (offset > 0) {
          firstPart = words.slice(0, offset);
          lastPart = words.slice(offset, words.length);
        }

        for (x=0; x < lastPart.length; x++) {
          if(x % n === 0) {
            lastPart[x] = lastPart[x].charAt(0).toUpperCase().concat(lastPart[x].substring(1));
          }
        }

        return firstPart.concat(lastPart).join(' ');
    }
